src/lib/lessons.ts

```diff
@@ -32,10 +58,12 @@ const lessonQuery = groq`
       'slug': slug.current,
       'type': 'lesson',
       'path': '/lessons/' + slug.current,
-      'title': title,
-      'duration': 0,
-      'thumb_url': null,
-      'media_url': *[_type=='videoResource' && _id == ^.resource->_id][0].hslUrl
+      title,
+      'thumb_url': thumbnailUrl,
+      resource->_type == 'videoResource' => {
+        'media_url': resource->hslUrl,
+        duration,
+      },
```

Comment:
this is sanity's conditional syntax. We put this in place to handle future resource types.